src/LibraryManager.Vsix/Contracts/HostInteraction.cs

```diff
@@ -53,43 +59,171 @@ public HostInteraction(string configFilePath, ILogger logger)
+        {
+            ProjectItem item = VsHelpers.DTE.Solution.FindProjectItem(filePath);
+            Project project = item?.ContainingProject;
+            bool deleteSucceeded = false;
+
+            if (project != null)
+            {
+                deleteSucceeded = await DeleteFileFromProject(item);
+            }
+            else
+            {
+                deleteSucceeded = await DeleteFileFromDisk(filePath);
+            }
+
+            if (deleteSucceeded)
+            {
+                Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleted, filePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
+            }
+            else
+            {
+                Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleteFail, filePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
+            }
 
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleted, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
+            return deleteSucceeded;
+        }
+        private Task<bool> DeleteFolderFromDisk (string folderPath)
+        {
+            return Task.Run(() =>
+            {
+                try
+                {
+                    DirectoryInfo directory = new DirectoryInfo(folderPath);
+                    if (directory.Exists && IsDirectoryEmpty(folderPath))
+                    {
+                        directory.Delete(true);
+                    }
+
+                    return true;
+                }
+                catch (Exception)
+                {
+                    return false;
                 }
-                catch (Exception ex)
+            });
+
+        }
+
+        private bool IsDirectoryEmpty(string path)
+        {
+            return !Directory.EnumerateFileSystemEntries(path).Any();
+        }
+
+        private Task<bool> DeleteFileFromDisk(string filePath)
+        {
+            return Task.Run(() =>
+            {
+                try
                 {
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleteFail, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
-                    Telemetry.TrackException("deletefilefailed", ex);
+                    FileInfo fileInfo = new FileInfo(filePath);
+                    if (fileInfo.Exists)
+                    {
+                        VsHelpers.CheckFileOutOfSourceControl(filePath);
+                        File.Delete(filePath);
+                    }
+
+                    return true;
                 }
+                catch (Exception)
+                {
+                    // Add telemetry here
+                    return false;
+                }
+            });
+        }
+
+        // TO DO: Move to VS helpers
+        private Task<bool> DeleteFileFromProject(ProjectItem projectItem)
```

Comment:
Move to VSHelpers: need to follow up with Bill to get it to properly work for all projects.